src/parser/lval.js

```diff
@@ -123,14 +123,11 @@ pp.parseBindingIdentifier = function () {
 };
 
 // Parses lvalue (assignable) atom.
-
 pp.parseBindingAtom = function () {
   switch (this.state.type) {
     case tt._yield:
-      if (this.state.strict || this.state.inGenerator) this.unexpected();
```

Comment:
There's another check inside `parseIdentifier/shouldAllowYieldIdentifier` via `checkReservedWord`.

Can revert it needed?